frontend/test/metabase/scenarios/admin/settings/public-sharing.cy.spec.js

```diff
@@ -0,0 +1,243 @@
+import { restore, modal, enableActionsForDB } from "__support__/e2e/helpers";
+import { SAMPLE_DB_ID } from "__support__/e2e/cypress_data";
+import { SAMPLE_DATABASE } from "__support__/e2e/cypress_sample_database";
+
+const { ORDERS_ID } = SAMPLE_DATABASE;
+
+describe("scenarios > admin > settings > public sharing", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+  });
+
+  it("should be able to toggle public sharing", () => {
+    cy.visit("/admin/settings/public-sharing");
+    cy.findByLabelText("Enable Public Sharing")
+      .should("be.checked")
+      .click()
+      .should("not.be.checked");
+  });
+
+  it("should see public dashboards", () => {
+    const expectedDashboardName = "Public dashboard";
+    const expectedDashboardSlug = "public-dashboard";
+    cy.createQuestionAndDashboard({
+      dashboardDetails: {
+        name: expectedDashboardName,
+      },
+      questionDetails: {
+        name: "Question",
+        query: {
+          "source-table": ORDERS_ID,
+        },
+      },
+    })
+      .then(({ body }) => {
+        const dashboardId = body.dashboard_id;
+        cy.wrap(dashboardId).as("dashboardId");
+        cy.request("POST", `/api/dashboard/${dashboardId}/public_link`, {});
+      })
+      .then(response => {
+        cy.wrap(response.body.uuid).as("dashboardUuid");
+      });
+
+    cy.visit("/admin/settings/public-sharing");
+
+    cy.findByText("Shared Dashboards").should("be.visible");
+    cy.findByText(expectedDashboardName).should("be.visible");
+    cy.get("@dashboardUuid").then(dashboardUuid => {
+      cy.findByText(
```

Comment:
@ranquild Now I check the whole link and click it and assert that the public link shows correctly too.